crates/optimism/payload/src/builder.rs

```diff
@@ -220,6 +220,26 @@ where
             None,
         ))
     }
+
+    fn cfg_and_block_env(
+        &self,
+        chain_spec: &ChainSpec,
+        parent: &Header,
+    ) -> (CfgEnvWithHandlerCfg, BlockEnv) {
+        let mut cfg_env = CfgEnvWithHandlerCfg::new_with_spec_id(CfgEnv::default(), SpecId::LATEST);
```

Comment:
this needs to be the same `CfgEnvWithHandlerCfg` as before, e.g. with the `is_optimism: true` and so on

Reply:
Similar comment to https://github.com/paradigmxyz/reth/pull/10510#discussion_r1731233532, `cfg_env` is mutated by `fill_cfg_env` with this being set https://github.com/paradigmxyz/reth/blob/72ed7a9b0812fb03b87be9e3877713dc06b7dc91/crates/optimism/evm/src/lib.rs#L108-L109

WDYT @onbjerg ?

Reply:
Since this is the optimism payload builder, I'd prefer we set `is_optimism` to be `true` regardless of what the chainspec says; this is needed for custom OP stack builds that do not necessarily have the same forks or network setup as any of the canonical OP stack setups. The `ChainSpec::is_optimism` function is kind of a leftover we should not rely on too heavily

Reply:
In that case, would it make more sense to set `is_optimism = true` in `OptimismEvmConfig::fill_cfg_env` rather than hard coding it in OptimismPayloadBuilder? Is there ever a case where we expect the cfg_env produced to _not_ be Optimism given its coming from OptimismEvmConfig?

Reply:
Same question for `spec_id` w.r.t to https://github.com/paradigmxyz/reth/pull/10510#discussion_r1730884340 since it's already set in fill_cfg_env

Reply:
yes, that would make sense - we should set `is_optimism = true` in `fill_cfg_env` for optimism.

Sorry, some of these code paths are sort of non-obvious, I keep forgetting that we mutate pretty much everything in `cfg_env` -.-